3.0 changelog (#961)
Total-RP/Total-RP-3: CHANGELOG.md

- Other players default profiles will no longer appear in your directory or be used for chat/nameplate customization.
- Character names are now customized in the TRP dice roll output.
- Glances with empty names will now show as such instead of replacing it with "...".
- Glances from your character still show their icons on your profile and target frame while disabled.
  - These will still not be sent to other players.
- Dragging a glance to reorder it will now display its icon on the cursor.
- Toolbar button and slash command for the NPC speech window now closes it if already shown.
- Clicking the target frame button for companion profiles now directly opens the profile.
  - The options menu to unbind/select a new profile can still be accessed with a right-click.
- The companion/mount profile buttons no longer change icon based on the selected profile.
- "Frames settings" has been split into Tooltip and Target frame categories.
  - The map scan button settings have been moved to the Directory category.
- Tooltip texts across the addon were reworked to follow a more consistent format and add clarity when needed.

> - Chat links can now be closed with Escape, and profile chat link tooltips have received minor display improvements.

Reviewer: ```suggestion
- Chat links can now be closed with Escape, and profile chat link tooltips have received minor display improvements.
- The Icon browser now displays icons in a larger size for optimal viewing pleasure.
```

Reviewer: or something to that effect

# Changelog version 3.0.0

This version adds support for patch 11.0.

Thanks to Raenore for the help on some of these features, and keyboardturner for the new silhouette logo.

## Added
- Added a new Relations menu located in the Customization section, alongside the moved Automation settings.
  - This menu lets you edit the preset relations you can have with other players, as well as create new ones.
  - Those relations are still only personal, and not shared with other players.
  - The relation will now be shown as a line on the tooltip in addition to the border color. This can be changed to show the relation description, or disabled in Tooltip settings.
- Added a "Currently" toolbar button letting you open a frame to quickly modify your Currently info.
- Added a toolbar button to open/close the main Total RP 3 window.
- Added a target frame button for your own companion to auto-fill the speech window with its name in brackets.
  - This will apply RP name customization to its speech lines.
- Added a dropdown to set whether you are open to other players walking up to your character.

>   - This will show as an icon on your tooltip.
>   - Existing tooltip bottom icons have also been changed for clarity.
>     - **TODO: INSERT IMAGE**

Reviewer 2: ```suggestion
  - This will show as an icon on your tooltip and in your profile characteristics.
  - Existing tooltip bottom icons have also been changed for clarity.
    ![image](https://github.com/user-attachments/assets/2ed353da-588a-45c4-b7af-fc8dfa7b9c74)
    - Book: Profile has an unread description.
    - Note: You have notes on this profile.
    - Three people: Walkup friendly.
```

This hopefully takes care of that, img below too:
![image](https://github.com/user-attachments/assets/2ed353da-588a-45c4-b7af-fc8dfa7b9c74)

- Added the ability to drag and drop personality traits to reorder them
  - Additional info now uses the same button style rather than drag and dropping the icon itself.
- Added the ability to duplicate a personality trait or additional info line, as well as convert preset lines into custom.
  - These are located within the new Options button at the end of the line, alongside deletion of the line.
  - Preset additional info lines appearing on the tooltip will no longer appear if converted to custom.
- Added indicators on additional info presets which will appear on tooltips by default.
- Added the ability to right-click an icon button to copy/paste icons between buttons or copy the name of the selected icon.
- Added a new background browser, as well as 33 new background options for the "About" section (and Extended documents).
- The toolbar will now be skinned while using ElvUI.
- Added X icons, X images and X musics from patch 11.0.

## Changed
- The UI has received an extensive makeover, including smoother scrolling in profiles list and directory.
- The toolbar and target frame buttons have had their icons updated to make their actions clearer.

>   - **TODO: INSERT IMAGE**

Reviewer: I guess images are going in the visual guide and as such this todo can be removed?

Thanks to Raenore for the help on some of these features, and keyboardturner for the new silhouette logo.

## Added
- Added a new Relations menu located in the Customization section, alongside the moved Automation settings.
  - This menu lets you edit the preset relations you can have with other players, as well as create new ones.
  - Those relations are still only personal, and not shared with other players.
  - The relation will now be shown as a line on the tooltip in addition to the border color. This can be changed to show the relation description, or disabled in Tooltip settings.
- Added a "Currently" toolbar button letting you open a frame to quickly modify your Currently info.
- Added a toolbar button to open/close the main Total RP 3 window.
- Added a target frame button for your own companion to auto-fill the speech window with its name in brackets.
  - This will apply RP name customization to its speech lines.
- Added a dropdown to set whether you are open to other players walking up to your character.
  - This will show as an icon on your tooltip and in your profile characteristics.
  - Existing tooltip bottom icons have also been changed for clarity.

>     ![image](https://github.com/user-attachments/assets/2ed353da-588a-45c4-b7af-fc8dfa7b9c74)

Reviewer 2: ```suggestion
```
On second thought, remove all imgs from the changelog and keep that to the visual guide.

- Tooltip texts across the addon were reworked to follow a more consistent format and add clarity when needed.
- Chat links can now be closed with Escape, and profile chat link tooltips have received minor display improvements.
- The Icon browser now displays icons in a larger size both on the grid and in tooltips.

> - Names and titles maximum length in tooltips has been reduced.

Reviewer: Should update this to also mention race and class. Also, separate line pointing out that we now show tooltips for more truncated fields like names, titles, and traits?
